frontend/src/lib/hooks/useOutsideClickHandler.ts

```diff
@@ -22,8 +22,8 @@ export function useOutsideClickHandler(
 
         if (allRefs.length > 0) {
             // Only attach event listeners if there's something to track
-            document.addEventListener('click', handleClick)
-            return () => document.removeEventListener('click', handleClick)
+            document.addEventListener('mouseup', handleClick)
+            return () => document.removeEventListener('mouseup', handleClick)
```

Comment:
`click` also fires on touch events. What implications does this change have for e.g. iPad users?

Reply:
good catch. I think it's correct to listen up "end" events in general so we probably want to also listen to touchend events to support mobile users.